circuit-chat.js

```diff
@@ -0,0 +1,280 @@
+const template = document.createElement('template');
+const initHtml = `
+<style>
+.name {
+  align-self: flex-start;
+  min-width: var(--name-width);
+}
+.item-content {
+  padding: var(--item-padding);
+  align-self: flex-start;
+}
+</style>
+<div part="main-container">
+  <div part="conversation-title" id="conversation-title"></div>
+  <div part="conversation-container">
+    <div id="conversation-feed"></div>
+  </div>
+  <input part="inp" type="text">
+  <button part="btn">Send</button>
+</div>`;
+template.innerHTML = initHtml;
+export class CircuitConversation extends HTMLElement {
+  // Attributes we care about getting values from.
+  static get observedAttributes() {
+    return ['convId'];
+  }
+
+  get connected() {
+    return this.hasAttribute('connected');
+  }
+
+  set _connected(isConnected) {
+    if (isConnected) {
+      this.setAttribute('connected', '');
+    } else {
+      this.removeAttribute('connected');
+    }
+  }
+
+  // Get the conversation here
+  set convId(value) {
+    // If no value given or if the conversation is the same as before
+    if (!value || !value.length || value === this._convId) {
+        return;
+    }
+    this._resetValues();
+    this._convId = value;
+    // Reflect to attribute
+    this._conversationFeed.display = 'none';
+    this._input.display = 'none';
+    this.setAttribute('convId', value);
+    this._getConversation();
+  }
+
+  set maxItems(value) {
+    // Max size of conversations store at once, defaults to 100
+    this._maxItems = value ? value : 100;
+  }
+
+  set initNumOfItems(value) {
+    // Initial number of items to retrieve, defaults to 20
+    this._initNumOfItems = value ? value : 20;
+  }
+
+  get convId() {
+    return this._convId;
+  }
+
+  get client() {
+    return this._client;
+  }
+
+  constructor() {
+    super();
+    this._client = null;
+
+    this.root = this.attachShadow({ mode: 'open' });
+    this.root.appendChild(template.content.cloneNode(true));
+
+    this._usersHashMap = {}; // Hashmap to store users names for the conversation feed
+    this._conversationFeed = this.root.getElementById('conversation-feed'); // Conversation Feed
+    this._conversationTitle = this.root.getElementById('conversation-title'); //  Title of the conversation
+    this._input = this.root.querySelector('input');
+    this._btn = this.root.querySelector('button');
+  }
+
+  // Delay Circuit initialization to speed up page load. This way the SDK
+  // can be loaded with 'async'
+  async _init() {
+    // this._conversationFeed.style.overflowY = 'auto';
+    this._client = Circuit.Client({
+      domain: this._domain,
+      client_id: this._clientId
+    });
+
+    // Replacing hr from circuit items to br
```

Comment:
Remove this. Instead define default CSS for hr to show a newline (or padding) instead